core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -64,6 +64,19 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
   @volatile private var dirtiestLogCleanableRatio = 0.0
   newGauge("max-dirty-percent", new Gauge[Int] { def value = (100 * dirtiestLogCleanableRatio).toInt })
 
+  /* a gauge for tracking the number of logCleaner runs */
+  @volatile private var numLogCleanRuns = 0
+  newGauge("num-runs", new Gauge[Int] { def value = numLogCleanRuns })
+
+  /* a gauge for tracking the last time of logCleaner run */
+  @volatile private var logCleanLastRun : Long = System.currentTimeMillis
```

Comment:
I agree, `time-since-last-run` is certainly a better metric as you explained. However, this stat is computed in the main loop of log cleaner thread. In order to capture `time-since-last-run`, I think we need another thread that runs a level above the log cleaner thread and it needs to update this metric periodically. Seems like it would add a whole lot of additional code and complexity, and in comparison, computation on the monitoring side might be less complex.